deepcell/image_generators/semantic.py

```diff
@@ -979,3 +980,878 @@ def random_transform(self, x, y=None, seed=None):
+                                    self.frame_shape[0],
+                                    self.frame_shape[1],
+                                    self.x.shape[4]))
+
+            if self.data_format == 'channels_first':
+                batch_y_semantic_list = [np.zeros(tuple([len(index_array),
+                                                         y_semantic.shape[1],
+                                                         self.frames_per_batch,
+                                                         self.frame_shape[0],
+                                                         self.frame_shape[1]]))
+                                         for y_semantic in self.y_semantic_list]
+            else:
+                batch_y_semantic_list = [
+                    np.zeros((len(index_array),
+                              self.frames_per_batch,
+                              self.frame_shape[0],
+                              self.frame_shape[1],
+                              y_semantic.shape[4]))
+                    for y_semantic in self.y_semantic_list
+                ]
+
+        else:
+            if self.data_format == 'channels_first':
+                batch_x = np.zeros((len(index_array),
+                                    self.x.shape[1],
+                                    self.frames_per_batch,
+                                    self.x.shape[3],
+                                    self.x.shape[4]))
+            else:
+                batch_x = np.zeros(tuple([len(index_array),
+                                          self.frames_per_batch] + list(self.x.shape)[2:]))
+
+            if self.data_format == 'channels_first':
+                batch_y_semantic_list = [np.zeros(tuple([len(index_array),
+                                                         y_semantic.shape[1],
+                                                         self.frames_per_batch,
+                                                         y_semantic.shape[3],
+                                                         y_semantic.shape[4]]))
+                                         for y_semantic in self.y_semantic_list]
+            else:
+                batch_y_semantic_list = [
+                    np.zeros(tuple([len(index_array), self.frames_per_batch] +
+                                   list(y_semantic.shape[2:])))
+                    for y_semantic in self.y_semantic_list
+                ]
+
+        for i, j in enumerate(index_array):
+            last_frame = self.x.shape[self.time_axis] - self.frames_per_batch
+
+            if last_frame == 0:
+                time_start = 0
+            else:
+                time_start = np.random.randint(0, high=last_frame)
+
+            time_end = time_start + self.frames_per_batch
+
+            if self.frame_shape:
+                last_row = self.x.shape[self.row_axis] - self.frame_shape[0]
+                last_col = self.x.shape[self.col_axis] - self.frame_shape[1]
```

Comment:
I'm wondering if we can generalize these `frame_shape` conditions to clean up some of these conditional blocks.